aiida_quantumespresso/workflows/pdos.py

```diff
@@ -0,0 +1,564 @@
+# -*- coding: utf-8 -*-
+"""Workchain to run Quantum ESPRESSO calculations that generate DoS and PDoS for a structure.
+
+This requires four computations:
+
+- SCF (pw.x), to generate the initial wavefunction.
+- NSCF (pw.x), to generate eigenvalues, generally with a denser k-point mesh and tetrahedra occupations.
+- Total DoS (dos.x), to generate total densities of state.
+- Partial DoS (projwfc.x), to generate partial densities of state, by projecting wavefunctions onto atomic orbitals.
+
+Additional functionality:
+
+- Setting ``'align_to_fermi': True`` in the input ``parameters`` node,
+  will ensure that the energy range is centred around the Fermi energy.
+
+Storage memory management:
+
+The wavefunction file(s) created by the nscf calculation can get very large (>100Gb).
+These files must be copied to dos and projwfc calculations, so storage memory limits can easily be exceeded.
+If this is an issue, setting the input ``serial_clean`` to ``True`` will not run these calculations in parallel,
+but instead run in serial and clean directories when they are no longer required:
+
+- Run the scf workchain
+- Run the nscf workchain, then clean the scf calculation directories
+- Run the dos calculation, then clean its directory
+- Run the projwfc calculation, then clean its directory
+
+Setting the input ``clean_workdir`` to ``True``, will clean any remaining directories,
+after the whole workchain has terminated.
+
+Also note that projwfc will fail if the scf/nscf calculations were run with a different number of procs/pools and
+``wf_collect=.false.`` (this setting is deprecated in newer version of QE).
+
+Related Resources:
+
+- `Electronic structure calculations user guide <https://www.quantum-espresso.org/Doc/pw_user_guide/node10.html>`_
+- `Density of States calculation blog <https://blog.levilentz.com/density-of-states-calculation/>`_
+- `Quantum ESPRESSO tutorial slides <http://indico.ictp.it/event/7921/session/320/contribution/1261/material/0/0.pdf>`_
+
+.. warning::
+
+    For QE v6.1, there is an issue using ``tetrahedra`` occupations, as is recommended for ``nscf``,
+    and both ``dos.x`` and ``projwfc.x`` will raise errors when reading the xml file
+    (see `this post <https://lists.quantum-espresso.org/pipermail/users/2017-November/039656.html>`_).
+
+"""
+from __future__ import absolute_import
+
+import jsonschema
+from six.moves import map
+
+from plumpy import ProcessSpec  # pylint: disable=unused-import
```

Comment:
is this needed?

Reply:
Don't think so, just missed this during my review. Will remove!